### PR TITLE
Integrate change from https://github.com/robinhood/faust/pull/701

### PR DESCRIPTION
**Pull request by wbarnha**
Intended to address https://github.com/faust-streaming/faust/issues/318.

### CODE REVIEW

**Comment by wbarnha** [comment]
I'm hesitant to merge this because after testing, we may not need this MR.

File: faust/transport/consumer.py [hunk 1]
```diff
@@ -1122,6 +1122,9 @@ def _new_offset(self, tp: TP) -> Optional[int]:
             batch = next(consecutive_numbers(acked))
             # remove them from the list to clean up.
             acked[: len(batch)] = []
+            latest_commit_id = self._committed_offset.get(tp, -1)
```
Comment (patkivikram) [inline faust/transport/consumer.py:1125]: is there a unit test to validate this change?

Comment (wbarnha) [reply faust/transport/consumer.py:1125]: I don't think so. I think we should write one to make sure this doesn't break functionality.

File: faust/transport/consumer.py [hunk 2]
```diff
@@ -1122,6 +1122,9 @@ def _new_offset(self, tp: TP) -> Optional[int]:
...
+            if acked and latest_commit_id and acked[0] <= latest_commit_id:
+                acked[:1] = []
```
Comment (patkivikram) [inline faust/transport/consumer.py:1127]: isn't this same as acked[0] = []

Comment (wbarnha) [reply faust/transport/consumer.py:1127]: Yeah, `acked[:1] = []` is the same as `acked[0] = []`, not sure why that syntax was chosen.

**Comment by wbarnha** [comment]
Unless anybody else can prove this MR is needed, I don't think we need to merge this. We already clear the acked buffer after rebalance. Closing.

**Pull request closed without merging**